### PR TITLE
Switch Flags to ENV variables for backend

### PR DESCRIPTION
**Pull request by brandon-red**
Changed all flags in `main.go` to env variables to address #710.

### CODE REVIEW

File: src/server/main.go [hunk 1]
```diff
@@ -32,46 +31,47 @@ func getPort() string {
 	return "8080"
 }
 
+// Check if env variable exists, if not set to default value, return value of env variable
+func checkEnv(key string, defaultVal string) string {
+	if os.Getenv(key) == "" {
+		os.Setenv(key, defaultVal)
```
Comment (KevinMackenzie) [inline src/server/main.go:37]: I'd say don't bother setting the env vars when using the default.  We should only be checking these on startup anyway.

File: src/server/main.go [hunk 2]
```diff
@@ -2,7 +2,6 @@ package main
 
 import (
 	"context"
-	"flag"
```
Comment (KevinMackenzie) [inline src/server/main.go:5]: Not mandatory, but it would be cool to have a `flag`-like interface that reads from env vars instead so we keep types.

Comment (brandon-red) [reply src/server/main.go]: Other than `bool` or `string`, are there other types that will need to be considered in the future? For now, the change I am implementing will handle only those two, but can be expanded on if need be.

Comment (KevinMackenzie) [reply src/server/main.go]: `int` and `uint` if you're up for it.  We probably don't care about differentiating between the different sizes (like `uint16` and `uint32`)

File: src/server/main.go [hunk 3]
```diff
@@ -32,48 +33,59 @@ func getPort() string {
...
-	googleAuthConfig := flag.String("google_auth", "", "<path-to-config>; Enables google sign-in API login")
-	noAuthConfig := flag.Bool("no_auth", false, "Enables username-only authentication for testing and development")
-	userCsifConfig := flag.String("interface", "sqlite", "The storage interface")
-	sqlitePathConfig := flag.String("sqlitePath", "sql/sqlite", "The path to the sqlite working directory")
-	dsEmulatorHost := flag.String("ds_emu_host", "", "The emulator host address for cloud datastore")
-	dsProjectId := flag.String("ds_emu_project_id", "", "The gcp project id for the datastore emulator")
-	ipAddressConfig := flag.String("ip_address", "0.0.0.0", "IP address of server")
-	portConfig := flag.String("port", "8080", "Port to serve application, use \"auto\" to select the first available port starting at 8080")
+	// Flag for noAuthConfig to keep boolean type
+	noAuthConfig := flag.Bool("no_auth", toBool(os.Getenv("no_auth")), "Enables username-only authentication for testing and development")
```
Comment (KevinMackenzie) [inline src/server/main.go:57]: I like this strategy.  Checking `flag`s first and fallback on the environment variables.  This should work in prod just fine and be convenient for back-end dev too.

**Comment by TGCrystal** [comment]
Not required, but I think environment variables are usually in all caps, so it wouldn't hurt for these ones to be all caps as well. An `OC_` prefix or something like that could also be nice to distinguish these as OpenCircuits specific.

**Comment by KevinMackenzie** [comment]
These changes got pulled into #1191

**Pull request closed without merging**